addons/frontend/index.html

```diff
@@ -15,6 +15,11 @@
     <style id="preview-theme-styles"></style>
     <!-- before all of code to avoid rerender by style replacement -->
     <script>
+      try {
+        // The right side of the equation will be string-replaced later and set in the main.js
+        window.vscode_state = JSON.parse(`VSCODE_STATE`);
+      } catch (ignored) {}
```

Comment:
A try catch to skipped error will make us lose all errors inside, like `JSON.parse` error or cases in which `vscode_state` doesn't exist. To detect whether frontend is opened in vscode, we can use:
```js
typeof acquireVsCodeApi !== 'undefined'
```